src/sexpr.rs

```diff
@@ -377,12 +377,25 @@ impl<'a> Lexer<'a> {
         }
     }
 
-    fn scan_symbol(&mut self, start: usize) -> &'a str {
+    /// Scan the current symbol and return the complete lexed string.
+    fn scan_symbol(&mut self, start: usize, is_quote: bool) -> &'a str {
+        // Are we within a || pair?
+        let mut quoted = is_quote;
         let mut end;
+
         loop {
             if let Some((ix, c)) = self.indices.peek() {
                 end = *ix;
-                if c.is_alphabetic() || c.is_numeric() || "~!@$%^&*_-+=<>.?/".contains(*c) {
+                if quoted && *c != '|' {
+                    // If we're in a quoted context, treat this as one identifier.
+                    self.indices.next();
+                    continue;
+                } else if *c == '|' {
+                    // If we see a quote, toggle the quoted flag.
+                    quoted = !quoted;
+                    self.indices.next();
+                    continue;
```

Comment:
Is it possible to escape the quotes? If not, this state machine seems reasonable to me 👍

Reply:
I don't *think* so? The quoted identifiers are specified as not containing `|` so I imagine you can't escape but I am also new to the feature.